lib/segment/src/vector_storage/dense/dynamic_mmap_flags.rs

```diff
@@ -68,9 +68,14 @@ impl fmt::Debug for DynamicMmapFlags {
 
 /// Based on the number of flags determines the size of the mmap file.
 fn mmap_capacity_bytes(num_flags: usize) -> usize {
-    let number_of_bytes = num_flags.div_ceil(u8::BITS as usize);
+    let mut number_of_bytes = num_flags.div_ceil(u8::BITS as usize);
 
-    max(MINIMAL_MMAP_SIZE, number_of_bytes.next_power_of_two())
+    // Exponentially grow file
+    number_of_bytes = max(MINIMAL_MMAP_SIZE, number_of_bytes.next_power_of_two());
+
+    // Bit slice uses multi-byte type as backing storage, so we must use a multiple of it
+    let align_bytes = BitSliceElement::BITS / u8::BITS;
+    number_of_bytes.next_multiple_of(align_bytes as usize)
 }
```

Comment:
Update mmap capacity bytes calculation here to ensure we always fit our `BitSlice` container type.

Reply:
Nit: I think it is not strictly unnecessary because of the previous line.
```rust
number_of_bytes = max(MINIMAL_MMAP_SIZE, number_of_bytes.next_power_of_two());
```
It will be always one of `0x100000`, `0x200000`, `0x400000`, `0x800000`, and so on, which is sure divisible by `size_of<usize>()`.

But if we want to be extra sure/explicit, I suggest to align to the same size (e.g. 8 bits) on all systems unconditionally.
So it won't be a problem when a file got created on a 32-bit machine and then transferred on 64-bit one.

Reply:
You're right. But I've just been cautious here in case we decrease the minimum size.

> But if we want to be extra sure/explicit, I suggest to align to the same size (e.g. 8 bytes) on all systems unconditionally.
So it won't be a problem when a file got created on a 32-bit machine and then transferred on 64-bit one.

This was currently achieved by using the u64 bit store type explicitly.

However, as it turns out we might need to use `u8` instead in which case this extra alignment isn't needed anymore.